tests/shadowcasting_test.cpp

```diff
@@ -407,199 +408,591 @@ static constexpr float O = LIGHT_TRANSPARENCY_SOLID;
 {
     // NOLINTNEXTLINE(cata-use-named-point-constants)
     grid_overlay test_case( { 1, 1 }, LIGHT_TRANSPARENCY_OPEN_AIR );
-    test_case.data = {
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, O, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
-        {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T}
+    test_case.data = { {
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, O, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T},
+            {T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T, T}
+        }
     };
 
     // NOLINTNEXTLINE(cata-use-named-point-constants)
     grid_overlay expected_results( { 1, 1 }, LIGHT_TRANSPARENCY_OPEN_AIR );
-    expected_results.data = {
-        {V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V},
-        {V, X, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V},
-        {V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V},
-        {V, V, V, V, V, O, O, O, V, V, V, V, V, V, V, V, V, V},
-        {V, V, V, V, V, V, O, O, O, O, O, O, O, V, V, V, V, V},
-        {V, V, V, V, V, V, V, O, O, O, O, O, O, O, O, O, O, O},
-        {V, V, V, V, V, V, V, V, O, O, O, O, O, O, O, O, O, O},
-        {V, V, V, V, V, V, V, V, V, O, O, O, O, O, O, O, O, O},
-        {V, V, V, V, V, V, V, V, V, V, O, O, O, O, O, O, O, O},
+    expected_results.data = { {
+            {V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V},
+            {V, X, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V},
+            {V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V, V},
+            {V, V, V, V, V, O, O, O, V, V, V, V, V, V, V, V, V, V},
+            {V, V, V, V, V, V, O, O, O, O, O, O, O, V, V, V, V, V},
+            {V, V, V, V, V, V, V, O, O, O, O, O, O, O, O, O, O, O},
+            {V, V, V, V, V, V, V, V, O, O, O, O, O, O, O, O, O, O},
+            {V, V, V, V, V, V, V, V, V, O, O, O, O, O, O, O, O, O},
+            {V, V, V, V, V, V, V, V, V, V, O, O, O, O, O, O, O, O},
+        }
     };
 
-    run_spot_check( test_case, expected_results );
+    run_spot_check( test_case, expected_results, true );
+    run_spot_check( test_case, expected_results, false );
```

Comment:
There's a Catch2 trick to make this into two test cases so the results are clearer, using `GENERATE`.  See #44216.